packages/logger/src/Logger.ts

```diff
@@ -876,6 +900,23 @@ class Logger extends Utility implements LoggerInterface {
       } else {
         this.#buffer.push([logLevel, [logLevel, input, extraInput]]);
       }
+    } else {
+      const xRayTraceId = this.envVarsService.getXrayTraceId() as string;
+
+      // Add the log item to the context
+      const context = this.#context[xRayTraceId] ?? [];
+      context.push([
```

Comment:
This is something I also touched upon in the original discussion, but for the sake of the review I'll include it here as well.

Having an unbounded buffer is a risk as you could run out of memory depending on how much and what you're logging.

For your specific use case this might not be a problem since you might know with relative certainty what you're logging and how much resources you're assigning to your Lambda functions, however from a library perspective this is a dangerous assumption to make.

If we were to implement this feature, we'd most likely want to set a max size for the buffer with a sensible default and an option to configure it. Depending on the performance impact of doing this, we might also want to consider calculating the size of each log and allow customers to set a max size in kb/mb for the buffer rather than a number of buffered logs, since the latter might have less meaning. Alternatively, or additionally, we might also want to consider implementing a sliding window mechanism that deletes older logs as newer are added and the buffer reaches the limit.

This above is the main reason why we haven't implemented the feature, as it requires some design and discussion.

Reply:
Yes, I agree! I think byte size and/or count limit should be implemented.

Since we store the stringified JSON (see comment on printLog/formatLog), we could easily calculate the UTF-8 byte size of each item and maintain a total byte size of the buffer. When a new log is added to the buffer and the total size exceeds the max allowed size, the buffer should drop the first N items until the total size is again below threshold.